tests/chainer_tests/initializer_tests/test_constant.py

```diff
@@ -92,13 +105,8 @@ def check_initializer(self, w):
             w, numpy.full(self.shape, self.fill_value),
             **self.check_options)
 
-    def test_initializer_cpu(self):
-        w = numpy.empty(self.shape, dtype=self.dtype)
-        self.check_initializer(w)
-
-    @attr.gpu
-    def test_initializer_gpu(self):
-        w = cuda.cupy.empty(self.shape, dtype=self.dtype)
+    def test_initializer(self, backend_config):
+        w = backend_config.xp.empty(self.shape, dtype=self.dtype)
```

Comment:
You can write `w = backend_config.get_array(numpy.empty(...))` otherwise you will get device mismatch error for CUDA device > 0.

Reply:
When using cuda_device:1 I am getting an internal error in cupy after changing this line to:
backend_config.get_array(numpy.empty(self.shape, dtype=self.dtype))

The error is as follows 
```
E   TypeError: an integer is required

  array[...] = 0
-----


cupy/core/core.pyx:498:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???

cupy/cuda/memory.pyx:476:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???

cupy/cuda/runtime.pyx:317:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   cupy.cuda.runtime.CUDARuntimeError: cudaErrorInvalidValue: invalid argument
```

But with cuda_device : 0 it works.

Am I missing a decorator?